samza-core/src/main/scala/org/apache/samza/system/chooser/BootstrappingChooser.scala

```diff
@@ -120,15 +129,12 @@ class BootstrappingChooser(
   override def register(systemStreamPartition: SystemStreamPartition, offset: String) {
     debug("Registering stream partition with offset: %s, %s" format (systemStreamPartition, offset))
 
-    // If the offset we're starting to consume from is the same as the upcoming 
-    // offset for this system stream partition, then we've already read all
-    // messages in the stream, and we're at head for this system stream 
-    // partition.
-    checkOffset(systemStreamPartition, offset, OffsetType.UPCOMING)
-
     wrapped.register(systemStreamPartition, offset)
 
-    registeredSystemStreamPartitions += systemStreamPartition
+    if (!registeredSystemStreamPartitions.contains(systemStreamPartition)
```

Comment:
Explained the reasoning in a comment and added a JIRA to comment as well. Please check if it would suffice.